src/test/java/TransportCommunicationIT.java

```diff
@@ -0,0 +1,155 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.opensearch.common.component.Lifecycle;
+import org.opensearch.common.network.NetworkAddress;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.transport.TransportAddress;
+import org.opensearch.test.OpenSearchIntegTestCase;
+import org.opensearch.transport.TransportService;
+import org.opensearch.transport.TransportSettings;
+import java.net.*;
+import java.io.*;
+
+import transportservice.RunPlugin;
+import transportservice.netty4.Netty4Transport;
+import org.opensearch.threadpool.ThreadPool;
+
+public class TransportCommunicationIT extends OpenSearchIntegTestCase {
+
+    private int port;
+    private Settings settings;
+    private final int minPort = 49152;
+    private final int maxPort = 65535;
+    private final String host = "127.0.0.1";
+    private volatile String clientResult;
+
+    @BeforeEach
+    public void setUp() {
+
+        // randomize port selection when creating settings
+        port = getRandomPort();
+
+        settings = Settings.builder()
+            .put("node.name", "node_extension_test")
+            .put(TransportSettings.BIND_HOST.getKey(), host)
+            .put(TransportSettings.PORT.getKey(), port)
+            .build();
+    }
+
+    @Test
+    public void testThatInfosAreExposed() {
+
+        RunPlugin runPlugin = new RunPlugin();
+        ThreadPool threadPool = new TestThreadPool("test");
+        Netty4Transport transport = runPlugin.getNetty4Transport(settings, threadPool);
+
+        // start netty transport and ensure that address info is exposed
+        try {
+            transport.start();
+            assertEquals(transport.lifecycleState(), Lifecycle.State.STARTED);
+
+            // check bound addresses
+            for (TransportAddress transportAddress : transport.boundAddress().boundAddresses()) {
+                assert (transportAddress instanceof TransportAddress);
+                assertEquals(host, transportAddress.getAddress());
+                assertEquals(port, transportAddress.getPort());
+            }
+
+            // check publish addresses
+            assert (transport.boundAddress().publishAddress() instanceof TransportAddress);
+            TransportAddress publishAddress = transport.boundAddress().publishAddress();
+            assertEquals(host, NetworkAddress.format(publishAddress.address().getAddress()));
```

Comment:
assertEquals syntax is assertEquals(expected, actual). Need to swap the arguments here and for the rest